ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/spot/provision/cucushift-installer-rehearse-aws-ipi-spot-provision-chain.yaml

```diff
@@ -0,0 +1,12 @@
+chain:
+  as: cucushift-installer-rehearse-aws-ipi-spot-provision
+  steps:
+  - ref: ipi-conf-aws
+  - ref: ipi-install-install
+  - ref: enable-qe-catalogsource
```

Comment:
Compared with the existing `cucushift-installer-rehearse-aws-ipi-provision` chain, it is:
```
  steps:
  - chain: ipi-conf-aws
  - chain: ipi-install
  - ref: enable-qe-catalogsource
```

Reply:
Thanks @jianlinliu , I modified and it seems to move ahead , but getting below error now - PTAL when time permits , any other place i can [check](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31682/rehearse-31682-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws/1565188089192648704) - 
`INFO[2022-09-01T04:04:56Z] Running step installer-rehearse-aws-ipi-install-install. 
INFO[2022-09-01T04:30:36Z] Logs for container test in pod installer-rehearse-aws-ipi-install-install: 
INFO[2022-09-01T04:30:36Z] Installing from release registry.build05.ci.openshift.org/ci-op-m9bxmk5f/release@sha[25](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31682/rehearse-31682-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws/1565188089192648704#1:build-log.txt%3A25)6:8ebe2496f3057b614be30e9ecc3c216ee634ca6d5ae0843403488e664dafb289
level=info msg=Credentials loaded from the "default" profile in file "/var/run/secrets/ci.openshift.io/cluster-profile/.awscred"
level=info msg=Consuming Install Config from target directory
level=info msg=Manifests created in: /tmp/installer/manifests and /tmp/installer/openshift
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 6113k  100 6113k    0     0  79.6M      0 --:--:-- --:--:-- --:--:-- 79.6M
Patched spotMarketOptions into /tmp/installer/openshift/99_openshift-cluster-api_worker-machineset-0.yaml
Patched spotMarketOptions into /tmp/installer/openshift/99_openshift-cluster-api_worker-machineset-1.yaml
Enabled AWS Spot instances for worker nodes
Will include manifests:
/tmp/secret/manifest_cluster-monitoring-config.yaml
level=warning msg=Found override for release image. Please be warned, this is not advised
level=info msg=Consuming Openshift Manifests from target directory
level=info msg=Consuming Worker Machines from target directory
level=info msg=Consuming Master Machines from target directory
level=info msg=Consuming OpenShift Install (Manifests) from target directory
level=info msg=Consuming Common Manifests from target directory
level=info msg=Credentials loaded from the "default" profile in file "/var/run/secrets/ci.openshift.io/cluster-profile/.awscred"
level=info msg=Creating infrastructure resources...
level=info msg=Waiting up to 20m0s (until 4:30AM) for the Kubernetes API at https://api.ci-op-m9bxmk5f-8f14f.qe.devcluster.openshift.com:6443...
level=info msg=Pulling VM console logs
level=info msg=Pulling debug logs from the bootstrap machine
level=error msg=Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get "https://api.ci-op-m9bxmk5f-8f14f.qe.devcluster.openshift.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 3.20.120.94:6443: connect: connection refused
level=error msg=Bootstrap failed to complete: Get "https://api.ci-op-m9bxmk5f-8f14f.qe.devcluster.openshift.com:6443/version": dial tcp 3.20.120.94:6443: connect: connection refused
level=error msg=Failed waiting for Kubernetes API. This error usually happens when there is a problem on the bootstrap host that prevents creating a temporary control plane.
level=error msg=The bootstrap machine failed to download the release image
level=info msg=Error: initializing source docker://registry.build05.ci.openshift.org/ci-op-m9bxmk5f/release@sha256:8ebe2496f3057b614be30e9ecc3c216ee634ca6d5ae0843403488e664dafb[28](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31682/rehearse-31682-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws/1565188089192648704#1:build-log.txt%3A28)9: reading manifest sha256:8ebe2496f[30](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31682/rehearse-31682-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws/1565188089192648704#1:build-log.txt%3A30)57b614be30e9ecc3c216ee634ca6d5ae0843403488e664dafb289 in registry.build05.ci.openshift.org/ci-op-m9bxmk5f/release: unauthorized: authentication required
level=info msg=Pull failed. Retrying registry.build05.ci.openshift.org/ci-op-m9bxmk5f/release@sha256:8ebe2496f3057b614be30e9ecc3c216ee634ca6d5ae0843403488e664dafb289...
level=info msg=Error: initializing source docker://registry.build05.ci.openshift.org/ci-op-m9bxmk5f/release@sha256:8ebe2496f3057b614be30e9ecc3c216ee634ca6d5ae0843403488e664dafb289: reading manifest sha256:8ebe2496f3057b614be30e9ecc3c216ee634ca6d5ae0843403488e664dafb289 in registry.build05.ci.openshift.org/ci-op-m9bxmk5f/release: unauthorized: authentication required
level=info msg=Bootstrap gather logs captured here "/tmp/installer/log-bundle-20220901043022.tar.gz"
Setup phase finished, prepare env for next steps
Copying log bundle...
Removing REDACTED info from log...
Copying required artifacts to shared dir
{"component":"entrypoint","error":"wrapped process failed: exit status 5","file":"k8s.io/test-infra/prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2022-09-01T04:30:[31](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/31682/rehearse-31682-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws/1565188089192648704#1:build-log.txt%3A31)Z"}
error: failed to execute wrapped command: exit status 5`

Reply:
Just as what I posted in the 1st comment, it is "chain: ipi-install", but not "ref: ipi-install-install"

Reply:
this we are trying to use for spot instances , I think if we use chain: ipi-install it will install non-spot workers .

Reply:
I did not see much change from your PR, beside adding `SPOT_INSTANCES` env. So do you mean "chain: ipi-install" can not inherit the `SPOT_INSTANCES` env setting?

Reply:
if I use it as chain , it fails as ipi-install-install chain doesn't exist error 
```
[vagrant@ibm-p8-kvm-03-guest-02 release]$ sudo make update 
make jobs
make[1]: Entering directory `/home/vagrant/release'
false || docker pull registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
latest: Pulling from ci/ci-operator-checkconfig
9bdecea3fbaa: Already exists 
a05dea350572: Already exists 
a093434cc8ca: Already exists 
632957bc3aa1: Already exists 
9e87b6d73963: Pull complete 
Digest: sha256:fb38715bc14c888209b3c5399eda747e300aadcaee8082c89344ab9b8de31728
Status: Downloaded newer image for registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
registry.ci.openshift.org/ci/ci-operator-checkconfig:latest
docker run --user 0:0 --platform linux/amd64 --rm -v "/home/vagrant/release/ci-operator/config:/ci-operator/config:z" -v "/home/vagrant/release/ci-operator/step-registry:/ci-operator/step-registry:z" registry.ci.openshift.org/ci/ci-operator-checkconfig:latest --config-dir /ci-operator/config --registry /ci-operator/step-registry
time="2022-09-01T03:36:18Z" level=fatal msg="failed to parse arguments" error="failed to load registry: Chain cucushift-installer-rehearse-aws-ipi-spot-provision contains non-existent chain ipi-install-install"
make[1]: *** [ci-operator-checkconfig] Error 1
```

Reply:
It is "chain: ipi-install", not "chain: ipi-install-install"!

Reply:
I suggest you can explore the chain/ref a bit more when similar issue hit next time.

```
$ find .|grep ipi-install-chain
./ci-operator/step-registry/ipi/install/ipi-install-chain.metadata.json
./ci-operator/step-registry/ipi/install/ipi-install-chain.yaml
$ cat ./ci-operator/step-registry/ipi/install/ipi-install-chain.yaml
chain:
  as: ipi-install
  steps:
  - ref: ipi-install-rbac
  - ref: openshift-cluster-bot-rbac
  - ref: ipi-install-install
  - ref: ipi-install-times-collection
  documentation: |-
    The IPI install step chain contains all the individual steps necessary to install an OpenShift cluster.
```